platform-core-module/src/main/java/com/softicar/platform/core/module/file/smb/jcifs/JcifsSmbClient.java

```diff
@@ -1,39 +1,47 @@
 package com.softicar.platform.core.module.file.smb.jcifs;
 
+import com.softicar.platform.common.core.exceptions.SofticarException;
 import com.softicar.platform.core.module.file.smb.ISmbClient;
 import com.softicar.platform.core.module.file.smb.ISmbDirectory;
 import com.softicar.platform.core.module.file.smb.ISmbFile;
 import com.softicar.platform.core.module.file.smb.SmbCredentials;
-import jcifs.Config;
-import jcifs.smb.NtlmPasswordAuthentication;
+import java.util.Properties;
+import jcifs.CIFSException;
+import jcifs.config.PropertyConfiguration;
+import jcifs.context.BaseContext;
+import jcifs.smb.NtlmPasswordAuthenticator;
 
 public class JcifsSmbClient implements ISmbClient {
 
-	@Override
-	public void initialize() {
-
-		disableDfs();
-	}
+	private static final BaseContext BASE_CONTEXT = setupBaseContext();
 
 	@Override
 	public ISmbFile createFile(String url, SmbCredentials credentials) {
 
-		var authentication = new NtlmPasswordAuthentication(credentials.getDomain(), credentials.getUsername(), credentials.getPassword());
-		return new JcifsSmbFile(url, authentication);
+		var cifsContext =
+				BASE_CONTEXT.withCredentials(new NtlmPasswordAuthenticator(credentials.getDomain(), credentials.getUsername(), credentials.getPassword()));
+		return new JcifsSmbFile(url, cifsContext);
 	}
 
 	@Override
 	public ISmbDirectory createDirectory(String url, SmbCredentials credentials) {
 
-		var authentication = new NtlmPasswordAuthentication(credentials.getDomain(), credentials.getUsername(), credentials.getPassword());
-		return new JcifsSmbDirectory(url, authentication);
+		var cifsContext =
+				BASE_CONTEXT.withCredentials(new NtlmPasswordAuthenticator(credentials.getDomain(), credentials.getUsername(), credentials.getPassword()));
+		return new JcifsSmbDirectory(url, cifsContext);
 	}
 
-	private void disableDfs() {
-
-		// FIXME vvvv explain why. don't make obscure claims. digging out #38419 should not be necessary to understand
-		// the reason. vvvv
-		// this is very important (see #38419)
-		Config.setProperty("jcifs.smb.client.dfs.disabled", "true");
+	private static BaseContext setupBaseContext() {
+
+		try {
+			// FIXME vvvv explain why. don't make obscure claims. digging out #38419 should not be necessary to understand
+			// the reason. vvvv
+			// this is very important (see #38419)
+			Properties jcifsProperties = new Properties();
+			jcifsProperties.setProperty("jcifs.smb.client.dfs.disabled", "true");
```

Comment:
Do we still need to do this? If we don't, we could finally get rid of that weird comment stack above.

Reply:
well, removed it and everything still worked, so im gonna remove it